### PR TITLE
feat: switch to using drop table procedure

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Switch to using drop table procedure
2. Remove unused code and metrics

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1727

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

**Comment by WenyXu** [comment]
PTAL @fengjiachun

File: src/frontend/src/instance/distributed.rs [hunk 1]
```diff
@@ -547,6 +515,30 @@ impl DistInstance {
...
+            task: DdlTask::new_drop_table(
+                table_name.catalog_name.to_string(),
+                table_name.schema_name.to_string(),
+                table_name.table_name.to_string(),
+                table_id,
+            ),
+        };
+
+        timeout(
+            // TODO(weny): makes timeout configurable.
```
Comment (killme2008) [inline src/frontend/src/instance/distributed.rs:533]: Lots of `timeout` paramaters are can't be configured right now in distribute mode, i think we should pay this tech debet ASAP. @MichaelScofield

File: src/common/meta/src/rpc/ddl.rs [hunk 2]
```diff
@@ -92,19 +106,17 @@ impl TryFrom<SubmitDdlTaskRequest> for PbSubmitDdlTaskRequest {
 
 pub struct SubmitDdlTaskResponse {
     pub key: Vec<u8>,
-    pub table_id: TableId,
+    pub table_id: Option<TableId>,
```
Comment (killme2008) [inline src/common/meta/src/rpc/ddl.rs:109]: Why did we change this type to `Option`? Looks unnecessary.

Comment (WenyXu) [reply src/common/meta/src/rpc/ddl.rs:109]: Currently, It only returns `table_id` when creating a table. For altering and dropping is no need to return the `table_id`.

File: src/frontend/src/instance/distributed.rs (hunk 1 above)
Comment (fengjiachun) [reply src/frontend/src/instance/distributed.rs:533]: We can provide a separate channel_manager for ddl-client and configure timeout independently.

**Review by fengjiachun — APPROVED** [review]

File: src/frontend/src/instance/distributed.rs (hunk 1 above)
Comment (MichaelScofield) [reply src/frontend/src/instance/distributed.rs:533]: tracked in #1935

**Pull request merged**